### ISSUE TITLE
EF Core 3.1 Beta 2 - OracleExceptions all returning as System.Exception: "Required user does not exist"

### ISSUE THREAD

**Issue by AdamWorley** [oracle/dotnet-db-samples#82]
It seems there may be an issue with how exceptions are being thrown where all excpetions are being thrown as `System.Exception: "Required user does not exist"`.

When running through tests to ensure that user friendly error messages are returned for different OracleException numbers, the tests are now failing when checking the error message.

When running the test using values that should work, the tests pass without any issues so it doesn't seem to be anything wrong in configuration.

**Comment by alexkeh** [comment]
When I test connecting with the beta 2 with a non-existent user, I receive an ORA-01017 invalid username/password error as the main exception. In your case, the .NET exception is appearing, which happens depending on the scenario. In my experience, the inner exception will contain the ORA error if it exists.

**Comment by AdamWorley** [comment]
![image](https://user-images.githubusercontent.com/28906234/87957154-d596cb00-caa7-11ea-8c08-1da8f629d1ca.png)

I am not getting an inner exception. The issue isn't the user id not being found, but internal errors being thrown as "User does not exsit".

If I provide an invalid user id as part of the connection string, I also receive ORA-01017 invalid username/password, so it seems like an issue that happens once authenticated and trying to run a command/query

**Comment by alexkeh** [comment]
I'll try and reproduce the error you are receiving. Can you provide details how you are generating the "Required user does not exist" error?

**Comment by AdamWorley** [comment]
@alexkeh yeah sure, thanks for looking into this.

When trying to call a stored procedure with checks in place on parameters, if a value is invalid the procedure will raise an error:
```sql
IF NOT V_FOUND_PWD THEN
    RAISE_APPLICATION_ERROR(-20999,'System not logged in') ;
END IF;
```

This worked in ef 2.* where an OracleException would be thrown and the error matches the above. However in 3.1 any of the Application Errors raised by the procedure always return as "Required user does not exist"

**Comment by alexkeh** [comment]
@AdamWorley 
Do you mind posting how you are propagating the error to EF Core? Using stored procedures from EF Core is anything but conventional. If you have a simple C# example of how your retrieving the error info and executing the stored procedure, that would be ideal. You can email the code if you prefer to dotnet_us(at)oracle.com.

**Comment by AdamWorley** [comment]
@alexkeh I have emailed the C# example, hopefully that will be helpful.

In 2.2 we were using `_dbContext.Database.ExecuteSqlCommand(query, parameters.ToArray());`, we're now using `_dbContext.Database.ExecuteSqlRaw(query, parameters.ToArray());`

This call is wrapped by a TryCatch that catches the OracleException and rethrows a relevant C# Exception message by using the Number property within the OracleException.

**Comment by alexkeh** [comment]
To update the community on this issue, it is a bug (31661791). It only occurs when using ODP.NET Core 2.19.80 as it can reproduced with either EF Core 2.x or 3.1 when using that provider version. The problem also only occurs with error code 20001. It does not reproduce with other error codes.

**Comment by AdamWorley** [comment]
@alexkeh Thank you again for your help

**Comment by alexkeh** [comment]
This bug has been fixed and will be available with the EF Core 3.1 production release.

**Comment by alexkeh** [comment]
Production release is now available on [NuGet Gallery](https://www.nuget.org/packages/Oracle.EntityFrameworkCore/3.19.80).
